pkg/resource/value.go

```diff
@@ -24,6 +24,11 @@ type Value struct {
 	bus minibus.Bus
 }
 
+// NewValue use by initialising with resource/Option.WithInitialValue(...).
+//
+// The reason for this is the go-protobuf package ( usage is proto.Clone() -> proto.Reset() -> proto.Merge() in this package )
+// doesn't support generics https://github.com/golang/protobuf/issues/1594#issuecomment-1978771310.
+// So without knowing in advance what the type of initialValue is, resource.Value can't change src from nil to some other type in order to proto.Merge into the dst value, when calling Value.Set.
```

Comment:
I think this comment needs a little work, it reads as very specific to your current frustrations with initial values rather than general documentation about the function.

- For a starter the first sentence isn't a well structured english sentence or descriptive of what the func is used for.
- Secondly, the WithInitialValue link isn't a GoDoc link.
- Thirdly I think it'd be better to remove the limitation in the code completely, why can't a resource.Value be created without an initial value and can we fix that.